docs/getting-started/integrations/angular.md

```diff
@@ -703,6 +703,45 @@ The `moduleResolution` option of the TypeScript configuration determines the alg
 * You can update Angular to version 18, where the `moduleResolution` option is set to `bundler`  by default.
 * You can import translations directly from our CDN, like: `import ‘https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/translations/es.umd.js’;`. This way, the editor will load the translations automatically, so you do not need to pass them manually into the config.
 
+### Potential issues with Jest testing
+
+Jest is the default test runner used by many Angular apps. Unfortunately, Jest does not use a real browser. Instead, it runs tests in Node.js with the use of JSDOM. JSDOM is not a complete DOM implementation and while it is apparently sufficient for standard apps, it is not able to polyfill all the DOM APIs that CKEditor&nbsp;5 requires.
+
+If this is not possible, you can use the following mocks to make the tests pass:
+
+```javascript
+beforeAll(() => {
+	window.scrollTo = jest.fn();
+
+	window.ResizeObserver = class ResizeObserver {
+		observe() {}
+		unobserve() {}
+		disconnect() {}
+	};
+
+	for (const key of ['InputEvent', 'KeyboardEvent']) {
+		window[key].prototype.getTargetRanges = () => {
+			const range = new StaticRange({
+				startContainer: document.body.querySelector('.ck-editor__editable p'),
+				startOffset: 0,
+				endContainer: document.body.querySelector('.ck-editor__editable p'),
+				endOffset: 0,
+			});
+
+			return [range];
+		};
+	}
+
+	Range.prototype.getClientRects = () => ({
+		item: () => null,
+		length: 0,
+		[Symbol.iterator]: function* () {},
+	});
+});
```

Comment:
```suggestion
beforeAll( () => {
	window.scrollTo = jest.fn();

	window.ResizeObserver = class ResizeObserver {
		observe() {}
		unobserve() {}
		disconnect() {}
	};

	for ( const key of [ 'InputEvent', 'KeyboardEvent' ] ) {
		window[ key ].prototype.getTargetRanges = () => {
			const range = new StaticRange( {
				startContainer: document.body.querySelector( '.ck-editor__editable p' ),
				startOffset: 0,
				endContainer: document.body.querySelector( '.ck-editor__editable p' ),
				endOffset: 0,
			} );

			return [ range ];
		};
	}

	Range.prototype.getClientRects = () => ( {
		item: () => null,
		length: 0,
		[ Symbol.iterator ]: function* () {},
	} );
} );
```

Reply:
I guess we want our code style in these snippets as well. I also might miss something - it's easy to mess up.